src/coreclr/jit/gentree.cpp

```diff
@@ -20431,8 +20452,16 @@ GenTree* Compiler::gtNewSimdBinOpNode(
                     if (varTypeIsLong(simdBaseType) || (simdBaseType == TYP_DOUBLE))
                     {
                         assert(varTypeIsSigned(simdBaseType));
-                        assert(compIsaSupportedDebugOnly(InstructionSet_AVX512F_VL));
-                        intrinsic = NI_AVX512F_VL_ShiftRightArithmetic;
+                        assert(compIsaSupportedDebugOnly(InstructionSet_AVX512F_VL) ||
```

Comment:
Rather than asserting that its `Avx512F_VL` or `AVX10v1`, we should just move the `assert(compIsaSupportedDebugOnly(InstructionSet_AVX512F_VL));` down into the else block (since `Avx10v1` is already caught by the main if branch)

Reply:
Made the changes apprpriately.